src/nms-api/IMessageProducer.cs

```diff
@@ -54,6 +58,48 @@ public interface IMessageProducer : System.IDisposable
 		/// </summary>
 		void Send(IDestination destination, IMessage message, MsgDeliveryMode deliveryMode, MsgPriority priority, TimeSpan timeToLive);
 
+                /// <summary>
+                /// Sends the message to the default destination for this producer
+                /// </summary>
+                void Send(IMessage message, CompletionListener completionListener);
```

Comment:
As a .NET developer, I find it a bit odd to see an asynchronous API designed with callbacks. I know that's the only way to do it in Java, but in my humble opinion, we should leverage what language has to offer. I don't have a strong opinion here though.

Reply:
its to keep in it inline with JMS API, i added additional .net way of async methods as you've seen as addition to the JMS methods.